src/common/imageio.c

```diff
@@ -708,14 +710,13 @@ int dt_imageio_export_with_flags(const int32_t imgid, const char *filename,
 
     GList *modules_used = NULL;
 
-    dt_dev_pop_history_items_ext(&dev, dev.history_end);
-
-    dt_ioppr_update_for_style_items(&dev, style_items, format_params->style_append);
+    dt_dev_pop_history_items_ext(&dev, appending ? dev.history_end : 0);
```

Comment:
At the end the fix was on a single line if we do not take into account cosmetic (but important for clarity) changes !

Reply:
yeah, i was also surprised how simple this turned out :-)